src/WebJobs.Script.Grpc/Extensions/ScriptInvocationContextExtensions.cs

```diff
@@ -9,13 +9,61 @@
 using Microsoft.Azure.WebJobs.Script.Description;
 using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
 using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Azure.WebJobs.Script.Grpc
 {
     internal static class ScriptInvocationContextExtensions
     {
-        public static async Task<InvocationRequest> ToRpcInvocationRequest(this ScriptInvocationContext context, ILogger logger, GrpcCapabilities capabilities)
+        private static async Task<RpcSharedMemory> SharedMemoryDataTransferAsync(ILogger logger, string invocationId, object value, ISharedMemoryManager sharedMemoryManager)
```

Comment:
rename to : `ToRpcSharedMemory` , Move `ToRpcSharedMemory` and any helpers to their own extensions class `SharedMemoryDataConversions`

Reply:
Mark this `internal` to allow for unit test